modules/client/src/internal/utils.ts

```diff
@@ -824,3 +863,763 @@ export function toPluginPermissionOperationType(
+          causes[0].concat(
+            await validateRevokeUpdatePluginPermissionAction(
+              actions[index],
+              pspAddress,
+              daoAddress,
+              graphql,
+            ),
+          );
+          break;
+        case ProposalActionTypes.APPLY_UPDATE:
+          const resCauses = await validateApplyUpdateFunction(
+            actions[index],
+            daoAddress,
+            graphql,
+            ipfs,
+          );
+          causes[0].concat(
+            resCauses,
+          );
+          break;
+      }
+    }
+    actions = actions.slice(3);
+    if (actions.length !== 0) {
+      const recCauses = await validateUpdatePluginProposalActions(
+        actions,
+        daoAddress,
+        pspAddress,
+        graphql,
+        ipfs,
+      );
+      causes = causes.concat(recCauses.causes);
+    }
+  } else if (isPluginUpdateActionWithRootPermission(classifiedActions)) {
+    causes[0] = [];
+    for (const [index, action] of classifiedActions.entries()) {
+      switch (action) {
+        case ProposalActionTypes.GRANT_PLUGIN_UPDATE_PERMISSION:
+          causes[0].concat(
+            await validateGrantUpdatePluginPermissionAction(
+              actions[index],
+              pspAddress,
+              daoAddress,
+              graphql,
+            ),
+          );
+          break;
+        case ProposalActionTypes.REVOKE_PLUGIN_UPGRADE_PERMISSION:
+          causes[0].concat(
+            await validateRevokeUpdatePluginPermissionAction(
+              actions[index],
+              pspAddress,
+              daoAddress,
+              graphql,
+            ),
+          );
+          break;
+        case ProposalActionTypes.GRANT_ROOT_PERMISSION:
+          causes[0].concat(
+            validateGrantRootPermissionAction(
+              actions[index],
+              daoAddress,
+              pspAddress,
+            ),
+          );
+          break;
+        case ProposalActionTypes.REVOKE_ROOT_PERMISSION:
+          causes[0].concat(
+            validateRevokeRootPermissionAction(
+              actions[index],
+              daoAddress,
+              pspAddress,
+            ),
+          );
+          break;
+        case ProposalActionTypes.APPLY_UPDATE:
+          causes[0].concat(
+            await validateApplyUpdateFunction(
+              actions[index],
+              daoAddress,
+              graphql,
+              ipfs,
+            ),
+          );
+          break;
+      }
+    }
+    actions = actions.slice(5);
+    if (actions.length !== 0) {
+      const recCauses = await validateUpdatePluginProposalActions(
+        actions,
+        daoAddress,
+        pspAddress,
+        graphql,
+        ipfs,
+      );
+      causes = [...causes, ...recCauses.causes];
+    }
+  } else {
+    causes[0] = [];
+    causes[0].push(
+      PluginUpdateProposalInValidityCause.INVALID_ACTIONS,
+    );
+    return {
+      isValid: false,
+      causes,
+    };
```

Comment:
either rewrite that the `else` is not needed or don't return here. because on the next line you start returning and it will return `isValid = false` because you set her the cause

Reply:
this check could also be moved up and check that the `actions` array is longer than 5 elements. because with the override on line 1603 the whole calculation process in the function is deleted and thus not necessary